arlo-client/src/components/utilities.ts

```diff
@@ -1,14 +1,21 @@
+import { toast } from 'react-toastify'
 import number from '../utils/number-schema'
+import { IErrorResponse } from '../types'
 
 export const api = async <T>(
   endpoint: string,
   options?: RequestInit
 ): Promise<T> => {
-  const res = await fetch(endpoint, options)
-  if (!res.ok) {
-    throw new Error(res.statusText)
+  try {
+    const res = await fetch(endpoint, options)
+    if (!res.ok) {
+      throw res
+    }
+    return res.json() as Promise<T>
+  } catch (err) {
+    const error = await err.json()
+    return Promise.reject(error)
```

Comment:
```suggestion
    throw error
```

Rather than mixing `async`/`await` with explicit `Promise`.